examples/preprocessing/plot_resample.py

```diff
@@ -0,0 +1,55 @@
+"""
+===============
+Resampling data
+===============
+
+Resampling can save memory and computation time. This example shows some
+approaches to resample data from 600 Hz to 100 Hz.
+"""
+# Authors: Marijn van Vliet <w.m.vanvliet@gmail.com>
+#
+# License: BSD (3-clause)
+#
+from __future__ import print_function
+
+import mne
+from mne.io import Raw
+from mne.datasets import sample
+
+print(__doc__)
+
+
+###############################################################################
+# The first approach is to resample the raw, continous data.
+
+data_path = sample.data_path()
+raw_fname = data_path + '/MEG/sample/sample_audvis_raw.fif'
+raw = Raw(raw_fname, preload=True)
+print('Original sampling rate:', raw.info['sfreq'], 'Hz')
+
+# Resample to 300 Hz 
+raw_resampled = raw.resample(300, copy=True)
+print('New sampling rate:', raw.info['sfreq'], 'Hz')
+
+###############################################################################
+# Because resampling also affects the stim channels, some trigger onsets might
+# be lost. While MNE attempts to downsample the stim channels in an intelligent
+# manner to avoid this, the recommended approach is to find events on the
+# original data before downsampling.
+print('Number of events before resampling:', len(mne.find_events(raw)))
+
+# Resample to 100 Hz (generates warning)
+raw_resampled = raw.resample(100, copy=True)
+print('Number of events after resampling:',
+      len(mne.find_events(raw_resampled)))
+
+# To avoid losing events, jointly resample the data and event matrix
+events = mne.find_events(raw)
+raw_resampled, events_resampled = raw.resample(100, events=events, copy=True)
+print('Number of events after resampling:', len(events_resampled))
+
+###############################################################################
+# Epoched data can be resampled as well.
```

Comment:
From a signal-processing perspective, what epoching *before* downsampling gets you is that you get to pull out data from the original data in what would end up being fractional sample locations of the downsampled raw signal. You could get something equivalent by downsampling your epochs, and then correcting for how far off you were by introducing a sub-sample delay (assuming you took care of the edges appropriately). So epoching then downsampling really is a better solution from a phase alignment perspective.